middle_end/flambda/from_lambda/cps_conversion.ml

```diff
@@ -988,12 +1205,15 @@ and cps_function env ~stub
   let body_cont = Continuation.create ~sort:Return () in
   let body_exn_cont = Continuation.create () in
   let free_idents_of_body = Lambda.free_variables body in
-  let body = cps_tail env body body_cont body_exn_cont in
   let exn_continuation : I.exn_continuation =
     { exn_handler = body_exn_cont;
       extra_args = [];
     }
   in
+  let new_env = Env.create ~current_unit_id:(Env.current_unit_id env)
```

Comment:
It's a bit surprising that `has_mutable` is not propagated, but given that it's never set to true anymore (and gets removed by your next PR) I guess it's fine.

Reply:
Should I instead remove it in this PR ?